code/game/machinery/autolathe/artist_bench.dm

```diff
@@ -0,0 +1,405 @@
+	if(..())
+		return
+
+	usr.set_machine(src)
+
+	if(href_list["oddity_name"])
+		if(oddity)
+			remove_oddity(usr)
+		else
+			insert_oddity(usr)
+		return TRUE
+
+	if(href_list["create_art"])
+		if(ishuman(usr))
+			var/mob/living/carbon/human/H = usr
+			var/ins_used = 0
+			if(H.stats.getPerk(PERK_ARTIST) && H.sanity.insight > 40)
+				ins_used = input("How much of your insight will you dedicate to this work? 40-[H.sanity.insight > 100 ? 100 : H.sanity.insight]","Insight Used") as null|num
+			else
+				ins_used = H.sanity.insight
+			create_art(ins_used, H)
+			return TRUE
+		return FALSE
+
+/obj/machinery/autolathe/artist_bench/proc/insert_oddity(mob/living/user, obj/item/inserted_oddity) //Not sure if nessecary to name oddity this way. obj/item/weapon/oddity/inserted_oddity
+	if(oddity)
+		to_chat(user, SPAN_NOTICE("There's already \a [oddity] inside [src]."))
+		return
+
+	if(!inserted_oddity && istype(user))
+		inserted_oddity = user.get_active_hand()
+
+	if(!istype(inserted_oddity))
+		return
+
+	if(!Adjacent(user) || !Adjacent(inserted_oddity))
+		return
+
+	GET_COMPONENT_FROM(C, /datum/component/inspiration, inserted_oddity)
+	if(!C || !C.perk)
+		return
+
+	if(istype(user) && (inserted_oddity in user))
+		user.unEquip(inserted_oddity, src)
+
+	inserted_oddity.forceMove(src)
+	oddity = inserted_oddity
+	inspiration = C
+	to_chat(user, SPAN_NOTICE("You set \the [inserted_oddity] into the model stand in [src]."))
+	SSnano.update_uis(src)
+
+/obj/machinery/autolathe/artist_bench/proc/remove_oddity(mob/living/user)
+	if(!oddity)
+		return
+
+	oddity.forceMove(drop_location())
+	to_chat(usr, SPAN_NOTICE("You remove \the [oddity] from the model stand in [src]."))
+
+	if(istype(user) && Adjacent(user))
+		user.put_in_hands(oddity)
+
+	oddity = null
+	inspiration = null
+	SSnano.update_uis(src)
+
+/obj/machinery/autolathe/artist_bench/proc/choose_base_art(ins_used, mob/living/carbon/human/user)
+	var/list/LStats = list()
+
+	if(inspiration && user.stats.getPerk(PERK_ARTIST))
+		LStats = inspiration.calculate_statistics()
+
+	var/weight_artwork_statue = 20
+	var/weight_artwork_revolver = 1 + LStats[STAT_VIG] * 2
+	var/weight_artwork_weapon = 1 + max(LStats[STAT_ROB], LStats[STAT_TGH]) * 2
+	var/weight_artwork_oddity = 1 + max(LStats[STAT_COG], LStats[STAT_BIO]) * 2
+	var/weight_artwork_tool = 2 + LStats[STAT_MEC] * 2
+	var/weight_artwork_toolmod = 2 + LStats[STAT_MEC] * 2
+	var/weight_artwork_gunmod = 2 + LStats[STAT_COG] * 2
+
+	if(ins_used >= 85)//Arbitrary values
+		weight_artwork_revolver += 9
+		weight_artwork_weapon += 9
+	if(ins_used >= 70)
+		weight_artwork_revolver += 4
+		weight_artwork_weapon += 4
+		weight_artwork_oddity += 13
+		weight_artwork_gunmod += 8
+	if(ins_used >= 55)
+		weight_artwork_gunmod += 4
+		weight_artwork_tool += 12
+		weight_artwork_toolmod += 12
+	else
+		weight_artwork_statue += 12
+
+	return pickweight(list(
+		//"artwork_revolver" = weight_artwork_revolver,	// OCCULUS EDIT - no meme guns
```

Comment:
Honestly, I wouldn't mind seeing guns come from this. But that might just be me. Why aren't the toolmods/gunmods and weapon options in here? Is this port finished Eris-side?

Reply:
At present, weapons, tools, and gunmods appear to be unimplemented, and also lacking in sprites.